src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,69 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+import { SortType } from '../../types/SortType';
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+type Props = {
+  query: string,
+  onQueryChange: (param: string) => void,
+  onSelectChange: (param: SortType) => void,
+  sortType: SortType,
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  query,
+  onQueryChange,
+  onSelectChange,
+  sortType,
+}) => {
+  const hadnleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    onSelectChange(+event.target.value as SortType);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onQueryChange(event.target.value);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            onChange={hadnleSelectChange}
+            value={sortType}
+          >
+            <option value={SortType.ALL}>All</option>
+            <option value={SortType.ACTIVE}>Active</option>
+            <option value={SortType.COMPLETED}>Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={handleInputChange}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {query.length !== 0 && (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              onClick={() => onQueryChange('')}
+            />
```

Comment:
```suggestion
            <button
              data-cy="clearSearchButton"
              type="button"
              className="delete"
              onClick={() => onQueryChange('')}
              aria-label="clear search'
            />
```
aria label attribute fix such eslint issue